testing/regress/ecl/spray_dir_test.ecl

```diff
@@ -0,0 +1,217 @@
+/*##############################################################################
+
+    Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+// The aim of this code is to test the wildcard spray from an empty and a not empty directory
+// and test:
+// 1. No failure if empty directory sprayed with wildcard
+// 2. 'failifnosourcefile' feature works as it suppose to be
+// 
+// When an empty directory sprayed with wildcard and 'failifnosourcefile' is enabled then 
+// it forces the DFU server to throw exeption.
+//
+// NOTHOR() effectively forces something to be executed globally.  At the moment if a global operation
+// fails then the query fails - rather than continuing and only failing if the result of using that
+// operation causes a failure.
+// So, to avoid to abort this code is excluded from Thor target
+//nothor
+
+//nohthor
+//class=spray
+
+
+import std.system.thorlib;
+import Std.File AS FileServices;
+import ^ as root;
+
+engine := thorlib.platform();
+prefix := '~regress::' + engine + '-';
+suffix := '-' + WORKUNIT;
+
+dropzonePath := '/var/lib/HPCCSystems/mydropzone/' : STORED('dropzonePath');
```

Comment:
Does this test pass during a make install regression run?
If not, is there an alternative here?

Reply:
Actually we have not mechanism to query drop zone path (only a [JIRA](https://track.hpccsystems.com/browse/HPCC-18819) for it), so in a "make install" system the  "-XdropzonePath=<realDropZonePath>" along with the (escaped) "localdir=$ESCAPED_TARGET_DIR::var::lib::^h^p^c^c^systems::mydropzone" should add to  ECL tool or Regression Test Engine parameter list and it will pass ([JIRA](https://track.hpccsystems.com/browse/HPCC-18822)).